File: Leetcode 1441. Build an Array With Stack Operations/lc 1441.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> ans;
        int ind=0;
        for(int i=1;i<=n && ind<target.size();i++){
            ans.push_back("Push");
            if(i<target[ind]){
                ans.push_back("Pop");
            }else{
                ind++;
            }
        }
        return ans;
    }
};

File: Leetcode 1441. Build an Array With Stack Operations/lc 1441.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> ans=new ArrayList<>();
        int ind=0;
        for(int i=1;i<=n && ind<target.length;i++){
            int val=target[ind];
            if(i<val){
                ans.add("Push");
                ans.add("Pop");
            }else if(i==val){
                ans.add("Push");
                ind++;
            }
            
        }
        return ans;
    }
}